src/utils/searchViewHelper.js

```diff
@@ -3,13 +3,19 @@ import {
   updateSearchValue,
   updateSearchResults,
   updateRepository,
+  updateDropdownError,
+  updateDropdownHelperText,
+  updateInputError,
+  updateInputHelperText,
 } from '../reducers/SearchSlice';
 import {
   fetchRecentRevisions,
   fetchRevisionByID,
   fetchRevisionsByAuthor,
 } from '../thunks/searchThunk';
 
+let _timeout = null;
+
 const searchViewHelper = {
```

Comment:
(not for this patch, more a surprise comment when looking at this code)
I was a bit surprised to see that this helper was touching the `store` directly. Is there a reason why this logic isn't in the react components directly?

Reply:
It was a decision to keep UI and logic separate, to avoid overly complex components (like we currently have in perfherder). Just a preference, I prefer it this way (I believe the others agreed with this also)